client/landing/gutenboarding/hooks/use-on-signup.ts

```diff
@@ -24,6 +25,12 @@ export default function useOnSignup() {
 	const newUser = useSelect( ( select ) => select( USER_STORE ).getNewUser() );
 	const newSite = useSelect( ( select ) => select( SITE_STORE ).getNewSite() );
 	const visibility = useNewSiteVisibility();
+	const anchorFmPodcastId = useAnchorFmQueryParam();
+	const isAnchorFmSignup = useSelect( ( select ) => select( ONBOARD_STORE ).getSelectedPodcastId() );
+
+	React.useEffect( () => {
+		dispatch( ONBOARD_STORE ).setPodcastId( anchorFmPodcastId );
```

Comment:
`anchorFmPodcastId` does this come direct from URL query params? Do we need/want to santize it?

Reply:
It does! Sanitizing it sounds like a good idea. 👍

Reply:
I added [a check for any non-numeric characters](https://github.com/Automattic/wp-calypso/pull/47600/commits/ed32e476902b24339280e308c0a23411ffe3e7c4).